src/new/wallet.js

```diff
@@ -2322,6 +2304,111 @@ class HathorWallet extends EventEmitter {
     const tx = await this.prepareCreateNewToken(name, symbol, amount, newOptions);
     return this.handleSendPreparedTransaction(tx);
   }
+
+  /**
+   * Identify all inputs from the loaded wallet
+   *
+   * @param {Transaction} tx The transaction
+   *
+   * @returns {{
+   * inputIndex: number,
+   * addressIndex: number,
+   * addressPath: string,
+   * }[]} List of indices and their associated address index
+   */
+   getWalletInputInfo(tx) {
+    storage.setStore(this.store);
+
+    const walletInputs = [];
+
+    for (const [inputIndex, input] of tx.inputs.entries()) {
+      const inputTx = this.getTx(input.hash);
+      if (!inputTx) {
+        // Input is not from our wallet
+        continue;
+      }
+      const utxo = inputTx.outputs[input.index];
+      if (utxo && this.isAddressMine(utxo.decoded.address)) {
+        const addressIndex = this.getAddressIndex(utxo.decoded.address);
+        // BIP32 address path
+        const addressPath = this.getAddressPathForIndex(addressIndex);
+        walletInputs.push({ inputIndex, addressIndex, addressPath });
+      }
+    }
+
+    return walletInputs;
+  }
+
+  /**
+   * Get signatures for all inputs of the loaded wallet.
+   *
+   * @param {Transaction} tx The transaction to be signed
+   * @param [options]
+   * @param {string} [options.pincode] PIN to decrypt the private key.
+   *                                   Optional but required if not set in this
+   *
+   * @returns {{
+   * inputIndex: number,
+   * addressIndex: number,
+   * addressPath: string,
+   * signature: string,
+   * pubkey: string,
+   * }} Input and signature information
+   */
+  getSignatures(tx, { pinCode = null } = {}) {
+    if (this.isFromXPub()) {
+      throw new WalletFromXPubGuard('getSignatures');
+    }
+    storage.setStore(this.store);
+    const pin = pinCode || this.pinCode;
+    if (!pin) {
+      throw new Error(ERROR_MESSAGE_PIN_REQUIRED);
+    }
+
+    // get private key
+    const accessData = storage.getItem('wallet:accessData');
+    const privateKeyStr = wallet.decryptData(accessData.mainKey, pin);
+    const key = HDPrivateKey(privateKeyStr);
+
+    const signatures = [];
+
+    for (const indices of this.getWalletInputInfo(tx)) {
+      const { addressIndex } = indices;
+      // Derive key to addressIndex
+      const derivedKey = key.deriveNonCompliantChild(addressIndex);
+      const privateKey = derivedKey.privateKey;
+      // Get tx signature and populate transaction
+      const sigDER = tx.getSignature(privateKey);
+      signatures.push({
+        signature: sigDER.toString('hex'),
+        pubkey: privateKey.publicKey.toString(),
```

Comment:
Is the default of this `toString` hex?

Reply:
Yes, it equivalent to `publicKey.toBuffer().toString('hex')`.

Reply:
Yes, it equivalent to `publicKey.toBuffer().toString('hex')`.